apollo-router/src/axum_factory/utils.rs

```diff
@@ -108,6 +121,94 @@ pub(super) async fn decompress_request_body(
     }
 }
 
+pub(super) async fn check_accept_header(
+    req: Request<Body>,
+    next: Next<Body>,
+) -> Result<Response, Response> {
+    let ask_for_html = req.method() == Method::GET && prefers_html(req.headers());
+
+    if req.headers().get(ACCEPT).is_some()
```

Comment:
```suggestion
    if accepts_wildcard(req.headers())
        || ask_for_html
        || accepts_multipart(req.headers())
        || accepts_json(req.headers()))
        {
                Ok(next.run(req).await)
    } else {
      Err((
            StatusCode::NOT_ACCEPTABLE,
            format!(
                r#"'accept' header is mandatory and can't be different than {:?}, {:?} or {:?}"#,
                APPLICATION_JSON_HEADER_VALUE,
                MULTIPART_DEFER_CONTENT_TYPE,
                APPLICATION_JSON_HEADER_VALUE
            ),
        )
            .into_response())
    }
```

it's a bit confusing with negative conditions (also it should have been something like `req.headers().get(ACCEPT).is_none() || (!accepts_wildcard(req.headers()) && ...)` because we should return an error when the `Accept` header is absent